src/middlewared/middlewared/plugins/system/debug.py

```diff
@@ -25,11 +26,24 @@ def debug_generate(self, job):
         Result value will be the absolute path of the file.
         """
         system_dataset_path = self.middleware.call_sync('systemdataset.config')['path']
-        execution_dir = get_debug_execution_dir(system_dataset_path)
-        dump = os.path.join(execution_dir, 'ixdiagnose.tgz')
 
-        # Be extra safe in case we have left over from previous run
-        shutil.rmtree(execution_dir, ignore_errors=True)
+        i = 0
+        while True:
+            execution_dir = get_debug_execution_dir(system_dataset_path, i)
```

Comment:
It doesn't look like you're specifying any permissions in `os.makedirs` call in `generate_debug() in ixdiagnose. You'll want some mechanism to set permissions so as to ensure that final path component here prevents unprivileged user from accessing the debug info.

Reply:
@anodos325 made those changes in https://github.com/truenas/ixdiagnose/pull/99